### PR TITLE
fix: update default sequence and docs

### PR DESCRIPTION
**Pull request by raymondfeng**
## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Sequence.md [hunk 1]
```diff
@@ -15,21 +15,72 @@ The contract of a `Sequence` is simple: it must produce a response to a request.
...
+      if (finished) {
+        // The response been produced by the middleware chain
+        return;
+      }
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+
+      debug('%s result -', route.describe(), result);
+      this.send(response, result);
```
Comment (bajtos) [inline docs/site/Sequence.md:81]: What happens when `this.send` returns a promise that will be eventually rejected? IMO, we should either keep `await` or return the send result directly.

```suggestion
      return this.send(response, result);
```

File: docs/site/Sequence.md [hunk 2]
```diff
@@ -15,21 +15,72 @@ The contract of a `Sequence` is simple: it must produce a response to a request.
...
     } catch (error) {
-      await this.reject(context, error);
+      this.reject(context, error);
```
Comment (bajtos) [inline docs/site/Sequence.md:83]: Ditto.

```suggestion
      return this.reject(context, error);
```

File: docs/site/Sequence.md [hunk 3]
```diff
@@ -71,6 +129,13 @@ class MySequence extends DefaultSequence {
 }
```
Comment (bajtos) [inline docs/site/Sequence.md:129]: Two lines above, you are calling `await this.send(...)`. Please make sure we are consistently showing the same approach in all places in our docs.

File: docs/site/Sequence.md (hunk 1 above)
Comment (raymondfeng) [reply docs/site/Sequence.md:81]: The signature of `Send` returns `void`:

```ts
export type Send = (response: Response, result: OperationRetval) => void;
```

File: docs/site/Sequence.md (hunk 2 above)
Comment (raymondfeng) [reply docs/site/Sequence.md:83]: `reject` returns `void` too.

```ts
export type Reject = (handlerContext: HandlerContext, err: Error) => void;
```

File: docs/site/Sequence.md (hunk 1 above)
Comment (bajtos) [reply docs/site/Sequence.md:81]: Interesting, I was remembering that both `Send` and `Reject` were returning a promise.

This change is fine then, just make sure to update other places for consistency please.

**Pull request merged**